packages/auth/AuthStore.ts

```diff
@@ -0,0 +1,126 @@
+// Recidiviz - a data platform for criminal justice reform
+// Copyright (C) 2022 Recidiviz, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+// =============================================================================
+
+import createAuth0Client, {
+  Auth0Client,
+  Auth0ClientOptions,
+  GetTokenSilentlyOptions,
+  User,
+} from "@auth0/auth0-spa-js";
+import { makeAutoObservable, runInAction } from "mobx";
+import qs from "qs";
+
+interface AuthStoreProps {
+  authSettings: Auth0ClientOptions | undefined;
+}
+export class AuthStore {
+  readonly authSettings: Auth0ClientOptions | undefined;
```

Comment:
why is it necessary to support this being undefined?

Reply:
Great question! I knew you'd spot this...

Early on in the JC app, we decided to have the backend send over the auth0 config and write them to the window object after hitting the endpoint (similar to case triage). This I believe was to handle the case where the backend either isn't running or for some reason fails in writing the config details to the window object. If it's undefined, we show an error page.

This is the `getAuthSettings` function in the RootStore:
```
const getAuthSettings = (): Auth0ClientOptions | undefined => {
  if (window.AUTH0_CONFIG) {
    return {
      domain: window.AUTH0_CONFIG.domain,
      client_id: window.AUTH0_CONFIG.clientId,
      redirect_uri: window.location.origin,
      audience: window.AUTH0_CONFIG.audience,
      useRefreshTokens: true,
    };
  }
  return undefined;
};
```

Reply:
I'm sure there's a better way for us to do this. 

If it ends up being completely unnecessary, happy to remove it from this package.